x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/grouping/Bucket.java

```diff
@@ -336,9 +333,18 @@ protected TypeResolution resolveType() {
                 : resolution.and(checkArgsCount(2)); // temporal amount
         }
         if (fieldType.isNumeric()) {
-            return bucketsType.isWholeNumber()
-                ? checkArgsCount(4).and(() -> isNumeric(from, sourceText(), THIRD)).and(() -> isNumeric(to, sourceText(), FOURTH))
-                : isNumeric(buckets, sourceText(), SECOND).and(checkArgsCount(2));
+            return isNumeric(buckets, sourceText(), SECOND).and(() -> {
+                if (bucketsType.isRationalNumber()) {
+                    return checkArgsCount(2);
```

Comment:
Do we have (verifier) test cases for when the 2nd argument is rational but we provide more than 2 arguments?

Reply:
I've added new tests around incorrect arguments, both in number and type, thanks.
(This should have been done in my earlier change already.)